### PR TITLE
Fix 500 error caused by deleted users on issues (#3675)

### PR DESCRIPTION
**Pull request by ghost**
This PR will resolve 500s caused by deleted users.

### CODE REVIEW

**Comment by LefsFlarey** [comment]
LGTM

**Comment by lunny** [comment]
hihi, I think one should not LGTM himself's PR.

**Comment by tboerger** [comment]
Needs to be rebased, otherwise LGTM

**Comment by tboerger** [comment]
Please reopen against master.

**Comment by bkcsoft** [comment]
@tboerger you can do that by editing the PR :wink: (found that out _today_ actually :laughing: https://github.com/blog/2224-change-the-base-branch-of-a-pull-request )

File: models/issue.go [hunk 1]
```diff
@@ -94,13 +94,8 @@ func (issue *Issue) loadAttributes(e Engine) (err error) {
 	if issue.Poster == nil {
 		issue.Poster, err = getUserByID(e, issue.PosterID)
 		if err != nil {
-			if IsErrUserNotExist(err) {
-				issue.PosterID = -1
-				issue.Poster = NewGhostUser()
-			} else {
-				return fmt.Errorf("getUserByID.(poster) [%d]: %v", issue.PosterID, err)
```
Comment (bkcsoft) [inline models/issue.go:101]: Please still print this error _if_ the User infact doesn't exist :slightly_smiling_face:

**Review by bkcsoft — DISMISSED** [review]

**Comment by strk** [comment]
An automated test would be great here

**Comment by strk** [comment]
@lunny can't lgtm check for self-lgtm ? (and I think your comment also counted)

**Comment by lunny** [comment]
lgtm can check but not the default. I have configed it correctly but the problem is lgtm don't know `not LGTM`.

**Comment by ghost** [comment]
@bkcsoft Would this work?

**Comment by lunny** [comment]
It seems it's no need this PR via `Files changed`.

**Comment by strk** [comment]
Having an automated test and instructions on how to reproduce the bug would help evaluating the need for change.

**Comment by lunny** [comment]
Yes

**Comment by ghost** [comment]
This is in reference to https://github.com/gogits/gogs/issues/3675

**Comment by strk** [comment]
I've just tried reproducing with these steps:
- Create a user X
- Have user X file issue on project Y
- Delete user X
- Visit issues of project Y
  Got the 505 internal error, before and after applying this patch.

Logs say:

```
2016/11/06 19:53:59 [...outers/repo/issue.go:188 Issues()] [E] Issues: LoadAttributes [42705]: user does not exist [uid: 3, name: ]
2016/11/06 19:53:59 [D] Template: status/500
```

**Comment by andreynering** [comment]
I tried this but it doesn't fix the error.

**Comment by ghost** [comment]
@andreynering @strk 

Sorry, took me awhile but this should fix the error completely, as I forgot to reset the `err` variable.

**Comment by strk** [comment]
The `lgtm` service is somewhat lame in that it doesn't reset upon pushing new commits.
@LefsFlarey: an automated test would still be great here, would you consider giving it a try ?

**Comment by strk** [comment]
Tested, now works, but the link to the ghost author is a 404 (maybe should be a page explaining what "Gost" is ? Or not be a link at all ?)

**Comment by strk** [comment]
Also, the "Ghost" user avatar seems to be the default avatar, maybe we want a Ghost-ed gopher ? :)

**Comment by strk** [comment]
btw, I noticed the user does _not_ turn into a link while in the Issue page, so it's only in the "Issues List" page that it needs be fixed.

(OT: the urls to user profile should really be changed IMHO - but it's for another discussion)

**Comment by lunny** [comment]
LGTM

**Pull request merged**